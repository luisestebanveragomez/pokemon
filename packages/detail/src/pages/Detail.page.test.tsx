import { render } from "@testing-library/react";

import DetailPage from "./Detail.page";
import { getDetailPokemon } from "../services/pokemon-detail.service";
import { usePokemonDetailContext } from "../contexts/pokemonDetail.Provider";

jest.mock("react-router-dom", () => ({
  __esModule: true,
  useNavigate: () => (key: string) => key,
  useParams: () => ({
    id: 12,
  }),
}));

jest.mock("../services/pokemon-detail.service", () => ({
  __esModule: true,
  getDetailPokemon: function (id: string) {
    return new Promise((resolve) => resolve({}));
  },
}));

jest.mock("../services/pokemon-detail.service", () => ({
  __esModule: true,
  getDetailPokemon: function (id: string) {
    return new Promise((resolve) => resolve({}));
  },
}));

jest.mock("react-i18next", () => ({
  __esModule: true,
  Trans: ({ i18nKey }) => i18nKey,
  useTranslation: () => ({
    t: (key: string) => key,
  }),
}));

jest.mock("../contexts/pokemonDetail.Provider", () => ({
  __esModule: true,
  usePokemonDetailContext: () => ({
    detail: {
      name: "test",
      type: "",
      stats: [],
    },
    setDetail: () => {},
  }),
}));

describe("Detail Page", () => {
  it("Should Render", () => {
    const { container } = render(<DetailPage />);
    expect(container).toMatchSnapshot();
  });
});
