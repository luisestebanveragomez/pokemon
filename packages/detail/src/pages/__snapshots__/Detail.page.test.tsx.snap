// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Page Should Render 1`] = `
<div>
  <section>
    <section>
      <a />
    </section>
    <section
      class="relative"
    >
      <figure
        class="pokemon"
      >
        <img
          alt=""
          class="pokemon__img"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/12.svg"
        />
        <figcaption>
          <span
            class="pokemon__category"
          />
          <h2
            class="pokemon__name"
          >
            test
          </h2>
        </figcaption>
      </figure>
      <div
        class="pokemon__number"
      >
        #
        12
      </div>
    </section>
    <section
      class="mt-4"
    >
      <h3
        class="stats__title"
      >
        pokemon.title
      </h3>
    </section>
  </section>
</div>
`;
